pyflakes/checker.py

```diff
@@ -310,10 +310,12 @@ class Binding(object):
                 the node that this binding was last used.
     """
 
-    def __init__(self, name, source):
+    def __init__(self, name, source, during_type_checking):
         self.name = name
         self.source = source
         self.used = False
+        assert isinstance(during_type_checking, bool)
+        self.during_type_checking = during_type_checking
```

Comment:
maybe `for_annotations` is a more descriptive name here?  (idk, I'm bikeshedding on this, feel free to ignore this suggestion, it just feels a little like the meaning of the name gets lost a bit)

Reply:
It's fine; takes a couple of seconds to change.